xarray/tests/test_backends.py

```diff
@@ -1275,20 +1304,22 @@ def test_dataset_caching(self):
 
     def test_open_mfdataset(self):
         original = Dataset({'foo': ('x', np.random.randn(10))})
-        with create_tmp_file() as tmp1:
-            with create_tmp_file() as tmp2:
-                original.isel(x=slice(5)).to_netcdf(tmp1)
-                original.isel(x=slice(5, 10)).to_netcdf(tmp2)
-                with open_mfdataset([tmp1, tmp2],
-                                    autoclose=self.autoclose) as actual:
-                    self.assertIsInstance(actual.foo.variable.data, da.Array)
-                    self.assertEqual(actual.foo.variable.data.chunks,
-                                     ((5, 5),))
-                    self.assertDatasetAllClose(original, actual)
-                with open_mfdataset([tmp1, tmp2], chunks={'x': 3},
-                                    autoclose=self.autoclose) as actual:
-                    self.assertEqual(actual.foo.variable.data.chunks,
-                                     ((3, 2, 3, 2),))
+        for _create_tmp_file in [create_tmp_file, create_tmp_file_pathlib]:
```

Comment:
Can you make a separate `test_open_mfdataset_path` test rather than squeezing this into `test_open_mfdataset`?

Some copy/paste is OK, but it doesn't need to all the check we have here around chunks and dask arrays. A simple `self.assertDatasetAllClose(original, actual)` would suffice.